src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/admission_control/AdmissionControlRca.java

```diff
@@ -0,0 +1,198 @@
+        AdmissionControlMetrics metricValue = new AdmissionControlMetrics();
+        metricValue.controllerName = controllerName;
+        metricValue.thresholdValue = (long) getAdmissionControlMetricValue(controllerName, admissionControlThresholdValue);
+        metricValue.currentValue = (long) getAdmissionControlMetricValue(controllerName, admissionControlCurrentValue);
+        return metricValue;
+    }
+
+    private HeapMetrics getHeapMetricModel() {
+        HeapMetrics heapMetrics = new HeapMetrics();
+        heapMetrics.usedHeap = getHeapMetricValue(heapUsedValue);
+        heapMetrics.maxHeap = getHeapMetricValue(heapMaxValue);
+        return heapMetrics;
+    }
+
+    @Override
+    public ResourceFlowUnit<HotNodeSummary> operate() {
+        long currentTimeMillis = System.currentTimeMillis();
+
+        AdmissionControlMetrics globalJVMMP = getAdmissionControlMetricModel(GLOBAL_JVMMP);
+        AdmissionControlMetrics requestSize = getAdmissionControlMetricModel(REQUEST_SIZE);
+        HeapMetrics heapMetrics = getHeapMetricModel();
+
+        if (heapMetrics.usedHeap == 0 || heapMetrics.maxHeap == 0) {
+            return new ResourceFlowUnit<>(currentTimeMillis);
+        }
+        double currentHeapPercent = (heapMetrics.usedHeap / heapMetrics.maxHeap) * 100;
+
+        LOG.debug("[AdmissionControl] controller={} currentValue={} thresholdValue={}",
+            GLOBAL_JVMMP, globalJVMMP.currentValue, globalJVMMP.thresholdValue);
+        LOG.debug("[AdmissionControl] controller={} currentValue={} thresholdValue={}",
+            REQUEST_SIZE, requestSize.currentValue, requestSize.thresholdValue);
+        LOG.debug("[AdmissionControl] usedHeap={} maxHeap={} currentHeapPercent={}",
+            heapMetrics.usedHeap, heapMetrics.maxHeap, currentHeapPercent);
+
+        if (requestSizeRangeConfiguration.hasRangeChanged(previousHeapPercent, currentHeapPercent)) {
+            double desiredThreshold = getThreshold(currentHeapPercent);
+            LOG.debug("[AdmissionControl] Observed range change. previousHeapPercent={} currentHeapPercent={} desiredThreshold={}",
+                previousHeapPercent, currentHeapPercent, desiredThreshold);
+            previousHeapPercent = currentHeapPercent;
+            ResourceContext context = new ResourceContext(Resources.State.UNHEALTHY);
+            InstanceDetails instanceDetails = getInstanceDetails();
+            HotNodeSummary nodeSummary = new HotNodeSummary(instanceDetails.getInstanceId(), instanceDetails.getInstanceIp());
+            HotResourceSummary resourceSummary = new HotResourceSummary(HEAP_MAX_SIZE, desiredThreshold, currentHeapPercent, 0);
+            nodeSummary.appendNestedSummary(resourceSummary);
+            RCA_VERTICES_METRICS_AGGREGATOR.updateStat(ADMISSION_CONTROL_RCA_TRIGGERED, instanceDetails.getInstanceId().toString(), 1);
+            return new ResourceFlowUnit<>(currentTimeMillis, context, nodeSummary);
+        }
```

Comment:
1. Let's javadoc string explaining when we mark this RCA as Unhealthy.
2. My understanding is that if the heap percentage range changes on the cluster, we will mark the RCA as Unhealthy. I have couple of questions here:
1/ What happens if the range change is a sink-in, heap usage went down. Will we mark this Unhealthy?
2/ We are not using the threshold here, why are we computing it?
3/ The `requestSize` metric is not used here. Is our AdmissionControl Unhealthy detection based only on JVM Heap Usage ? If yes. then why are we consuming this metric in analysis graph ?
4/ What happens if the node has index block due to disk space and JVM Heap Usage is low. How will the RCA behave and what action will the decider take?

Reply:
Added comments in code